pkg/asset/installconfig/gcp/client.go

```diff
@@ -370,32 +370,30 @@ func (c *Client) getServiceUsageService(ctx context.Context) (*serviceusage.Serv
 	return svc, nil
 }
 
-// GetServiceAccounts retrieves a list of service accounts that are valid for a project.
-func (c *Client) GetServiceAccounts(ctx context.Context, project string) ([]string, error) {
+// GetServiceAccount retrieves a list of service accounts that are valid for a project.
+func (c *Client) GetServiceAccount(ctx context.Context, project, serviceAccount string) (string, error) {
 	ctx, cancel := context.WithTimeout(ctx, 1*time.Minute)
 	defer cancel()
 
 	svc, err := iam.NewService(ctx)
 	if err != nil {
-		return nil, errors.Wrapf(err, "failed create IAM service")
+		return "", errors.Wrapf(err, "failed create IAM service")
 	}
 
-	rsp, err := svc.Projects.ServiceAccounts.List(fmt.Sprintf("projects/%s", project)).Do()
+	rsp, err := svc.Projects.ServiceAccounts.Get(fmt.Sprintf("projects/%s/serviceAccounts/%s", project, serviceAccount)).Do()
 	if err != nil {
-		return nil, errors.Wrapf(err, fmt.Sprintf("failed to list service accounts for project %s", project))
+		return "", errors.Wrapf(err, fmt.Sprintf("failed to find service account %s in project %s", serviceAccount, project))
 	}
 
-	serviceAccounts := []string{}
-	for _, act := range rsp.Accounts {
-		// The service accounts are listed in the format:
-		// projects/{project-id}/serviceAccounts/{ServiceAccountName}
-		// The use of the service account is by the name only when passed through manifests, so only
-		// the service account name is pulled here
-		saSplit := strings.Split(act.Name, "/")
-		// grab the last element no matter the length of the split list
-		serviceAccounts = append(serviceAccounts, saSplit[len(saSplit)-1])
+	// The service accounts are listed in the format:
+	// projects/{project-id}/serviceAccounts/{ServiceAccountName}
+	// The use of the service account is by the name only when passed through manifests, so only
+	// the service account name is pulled here
+	saSplit := strings.Split(rsp.Name, "/")
+	if len(saSplit) == 0 {
+		return "", fmt.Errorf("no service account name found")
 	}
-	return serviceAccounts, nil
+	return saSplit[len(saSplit)-1], nil
```

Comment:
Hm I'm not sure I understand why we are splitting the name like this and returning this value. Do we need to do this?